### PR TITLE
Add Local Filesystem Secret Backend (v1-10)

### PR DESCRIPTION
**Pull request by mik-laj**
Backport: https://github.com/apache/airflow/pull/8436
---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Target Github ISSUE in description if exists
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/exceptions.py [hunk 1]
```diff
@@ -117,3 +121,34 @@ class DagConcurrencyLimitReached(AirflowException):
...
+
+
+file_syntax_error = namedtuple('FileSyntaxError', 'line_no message')
+"""Information about a single error in a file."""
+
+
+class AirflowFileParseException(AirflowException):
+    """
+    Raises when connection or variable file can not be parsed
+    :param msg: The human-readable description of the exception
```
Comment (kaxil) [inline airflow/exceptions.py]:
```suggestion
    Raises when connection or variable file can not be parsed

    :param msg: The human-readable description of the exception
```

File: airflow/utils/code_utils.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+def prepare_code_snippet(file_path, line_no, context_lines_count=5):
+    """
+    Prepare code snippet with line numbers and  a specific line marked.
```
Comment (kaxil) [inline airflow/utils/code_utils.py]:
```suggestion
    Prepare code snippet with line numbers and a specific line marked.

```

**Comment by mik-laj** [comment]
@kaxil Github Action is green. Can I ask for review?

**Review by kaxil — APPROVED** [review]
That looks good, thanks for the backport PR @mik-laj , appreciate it :)

**Pull request merged**